### ISSUE TITLE
Incorrect Inspection for unused

### ISSUE THREAD

**Issue by TheEzraMiller** [kalessil/phpinspectionsea#1940]
| Subject        | Details                                                                       |
| :------------- | :---------------------------------------------------------------------------- |
| Plugin         | Php Inspections (EA Extended) |
| Language level | PHP 7.2                  |

Current behaviour
![Screenshot 2024-04-10 at 08 53 39](https://github.com/kalessil/phpinspectionsea/assets/2478416/930d3e79-ee69-4ddb-9b24-985176b2b69e)
See sample code $var1 on line 8 is actually used if the if conditions are met hence it shouldn't be highlighted as unused by inspections

**Comment by ea-inspections-team** [comment]
Thank you for reporting @TheEzraMiller!

We are confirming the bug, and able to reproduce with the following example:
```php
function example(bool $condition) {
    if ($condition) {
        $dynamic = '';
        $dynamic .= ''; // <-- [EA] Parameter/variable is overridden, but is never used or appears outside of the scope.
    }

    return $dynamic ?? null;
}
```

**Comment by ea-inspections-team** [comment]
Thank you for reporting @TheEzraMiller!
The false positive has been resolved.
